### PR TITLE
Weaviate backend

### PR DESCRIPTION
**Pull request by Tarraann**
### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [x] I have read the contributing guide and my code conforms to the guidelines.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have added the required tests.

### CODE REVIEW

File: superagi/vector_store/weaviate.py [hunk 1]
```diff
@@ -56,46 +53,48 @@ def __init__(
...
+            metadata = metadatas[i] if metadatas else {}
+            data_object = metadata.copy()
+            data_object[self.text_field] = text
+            vector = self.embedding_model.get_embedding(text)
+            id = str(uuid4())
+            result = {"ids": id, "data_object": data_object, "vectors": vector}
+            collected_ids.append(id)
+            try:
+                self.add_embeddings_to_vector_db(result)
+            except:
```
Comment (TransformerOptimus) [inline superagi/vector_store/weaviate.py:68]: no need to raise this again here.

File: superagi/tools/knowledge_search/knowledge_search.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def _execute(self, query: str):
         embedding_model = AgentExecutor.get_embedding(model_source, model_api_key)
         try:
             if vector_db_index.state == "Custom":
-                filters = {}
+                filters = None
```
Comment (TransformerOptimus) [inline superagi/tools/knowledge_search/knowledge_search.py:52]: why do we need it?

Comment (Tarraann) [reply superagi/tools/knowledge_search/knowledge_search.py:52]: Sir this is for metadata filtering, for marketplace knowledge we have a default filter on knowledge name but for custom currently we are not giving user the option to specify filters

File: superagi/models/vector_db_indices.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def delete_vector_db_index(cls, session, vector_index_id):
         session.commit()
 
     @classmethod
-    def add_vector_index(cls, session, index_name, vector_db_id, dimensions, state):
+    def add_vector_index(cls, session, index_name, vector_db_id, state, dimensions = None):
```
Comment (CaptainLevi0007) [inline superagi/models/vector_db_indices.py:50]: Add comment as discussed.

File: superagi/vector_embeddings/weaviate.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+class Weaviate(VectorEmbeddings):
+    
+    def __init__(self, uuid, embeds, metadata):
+        self.uuid = uuid
+        self.embeds = embeds
+        self.metadata = metadata
+    
+    def get_vector_embeddings_from_chunks(self):
+        """ Returns embeddings for vector dbs from final chunks"""
+        result = {}
```
Comment (CaptainLevi0007) [inline superagi/vector_embeddings/weaviate.py:13]: result = {'ids': self.uuid, 'data_object': self.metadata, 'vectors': self.embeds}

**Pull request merged**